x-pack/plugins/security_solution/server/fleet_integration/handlers/create_event_filters.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import uuid from 'uuid';
+import { i18n } from '@kbn/i18n';
+import { ENDPOINT_EVENT_FILTERS_LIST_ID } from '@kbn/securitysolution-list-constants';
+import type { Logger } from '@kbn/core/server';
+import type { PackagePolicy } from '@kbn/fleet-plugin/common';
+import type { ExceptionListClient } from '@kbn/lists-plugin/server';
+import type { EventFilters } from '../types';
+
+const PROCESS_INTERACTIVE_ECS_FIELD = 'process.entry_leader.interactive';
+
+/**
+ * Check if Event Filter list exists and Create Event Filters for the Elastic Defend integration.
+ */
+export const createEventFilters = async (
+  logger: Logger,
+  exceptionsClient: ExceptionListClient,
+  eventFilters: EventFilters,
+  packagePolicy: PackagePolicy
+) => {
+  const exceptionList = await exceptionsClient.getExceptionList({
+    id: undefined,
+    listId: ENDPOINT_EVENT_FILTERS_LIST_ID,
+    namespaceType: 'agnostic',
+  });
+
+  if (exceptionList == null) {
```

Comment:
Hey @dasansol92, thank you for pointing this out. I just checked it, and It looks like the [EventFiltersApiClient](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/management/pages/event_filters/service/api_client.ts) is located on the client side, however, the `getPackagePolicyPostCreateCallback` that calls `createEventFilters` executes on the server side.

So, one possible strategy could be to call the `EventFiltersApiClient` from the UI on the [EndpointPolicyCreateExtension](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/management/pages/policy/view/ingest_manager_integration/endpoint_policy_create_extension.tsx) when the component is first mounted? That would ensure the list is created by the moment the callback is executed; However, this null check on the server function would still be needed in case the function is called apart.

Reply:
Sorry, you are right. I didn't figured out this was server code.